x-pack/plugins/siem/public/components/open_timeline/index.tsx

```diff
@@ -261,4 +308,160 @@ export class StatefulOpenTimeline extends React.PureComponent<Props, State> {
       selectedItems: [],
     });
   };
+
+  private openTimeline: OnOpenTimeline = ({
+    duplicate,
+    timelineId,
+  }: {
+    duplicate: boolean;
+    timelineId: string;
+  }) => {
+    const {
+      addNotes,
+      addTimeline,
+      closeModalTimeline,
+      isModal,
+      setTimelineRangeDatePicker,
+      updateIsLoading,
+    } = this.props;
+
+    if (isModal && closeModalTimeline != null) {
+      closeModalTimeline();
+    }
+
+    updateIsLoading({ id: 'timeline-1', isLoading: true });
+    this.props.apolloClient
+      .query<GetOneTimeline.Query, GetOneTimeline.Variables>({
+        query: oneTimelineQuery,
+        fetchPolicy: 'no-cache',
+        variables: { id: timelineId },
+      })
+      .then(result => {
+        const timelineToOpen: TimelineResult = omitTypenameInTimeline(
+          getOr({}, 'data.getOneTimeline', result)
+        );
+        const { notes, ...timelineModel } = timelineToOpen;
+        const momentDate = dateMath.parse('now-24h');
+        setTimelineRangeDatePicker({
+          from: getOr(momentDate ? momentDate.valueOf() : 0, 'dateRange.start', timelineModel),
+          to: getOr(Date.now(), 'dateRange.end', timelineModel),
+        });
+        addTimeline({
+          id: 'timeline-1',
+          timeline: {
+            ...assign(this.props.timeline, timelineModel),
```

Comment:
I would really strongly suggest breaking some of these out into separate functions. This is very tricky to read through and no unit tests for them.